src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -108,5 +108,50 @@ private static string LoadRules()
                     Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location),
                     "Rules/BuiltInRules.json"));
         }
+
+        /// <summary>
+        /// Loads a configurations file. If no file was passed, checks the default directory for this file.
+        /// </summary>
+        /// <param name="configurationsFilePath">The path to a configuration file to read.</param>
+        /// <returns>Configuration file path contents if a file exists.</returns>
+        private string GetConfigurationFileContents(FileInfo configurationsFilePath)
+        {
+            try
+            {
+                string configurationFileContents = configurationsFilePath == null ? null : File.ReadAllText(configurationsFilePath.FullName);
+                if (configurationFileContents == null)
+                {
+                    var defaultPath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location),
+                        "/Configurations/Configuration.json");
+                    if (File.Exists(defaultPath))
+                    {
+                        Console.WriteLine(Environment.NewLine + Environment.NewLine + $"Configurations File: {defaultPath}");
+                        return File.ReadAllText(defaultPath);
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                }
+
+                Console.WriteLine(Environment.NewLine + Environment.NewLine + $"Configurations File: {configurationsFilePath}");
+                return configurationFileContents;
```

Comment:
If you change what the `if` statements are looking for above, you can group this code with the relevant logic reading the file.
For example:
``` cs
if (configurationsFilePath != null)
{
    string configurationFileContents = File.ReadAllText(configurationsFilePath.FullName);
    // print file name and return contents
}

// continue in default scenario
var defaultPath = ...
if (exists)
{
    // print name and return contents
}

return null;
```

Reply:
I re-ordered it.